### PR TITLE
[DS-3795] Manage versions of some buggy transitive dependencies. (for master)

### PR DESCRIPTION
**Pull request by terrywbrady**
Port https://github.com/DSpace/DSpace/pull/2082 to master

### CODE REVIEW

File: dspace-spring-rest/pom.xml [hunk 1]
```diff
@@ -155,6 +155,12 @@
                 <artifactId>spring-expression</artifactId>
                 <version>${spring.version}</version>
             </dependency>
+            <!-- Solr introduces multiple versions of zookeeper. So, we specify the version we want -->
```
Comment (mwoodiupui) [inline dspace-spring-rest/pom.xml]: See dspace-solr/pom.xml.  If we now have two projects dependent on zookeeper, then we should pull both of these dependency management declarations up into a single one in parent.

**Review by mwoodiupui — CHANGES_REQUESTED** [review]
Please merge duplicate zookeeper dependencies from dspace-solr and dspace-spring-rest up to dspace-parent.  Otherwise LGTM.

File: pom.xml [hunk 2]
```diff
@@ -1505,7 +1505,7 @@
             <dependency>
                 <groupId>com.fasterxml.jackson.core</groupId>
                 <artifactId>jackson-annotations</artifactId>
-                <version>2.8.11.1</version> <!-- TODO sync. with jackson.version -->
```
Comment (mwoodiupui) [inline pom.xml]: Wait, it looks like we are still using Jackson 2.8.11?  Then don't change this.  jackson-annotations-2.8.11 is buggy.  There is no 2.8.11.1 version of the other Jackson artifacts.  Sorry, thought we had upgraded the rest of Jackson.

**Comment by tdonohue** [comment]
@mwoodiupui : duplicate dependencies were moved up to Parent POM

**Review by mwoodiupui — APPROVED** [review]

**Pull request merged**